Next to Dev

microhobby opened the pull request.

microhobby committed:
dotnetAvalonia: update vscode extensions

Let's use the community extension for Avalonia, it has a fix that
prevents the extension from crashing.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetAvalonia: tasks: Trim trailing spaces

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
gambasForms: update docker-compose to use commontorizon/weston:3.3.0

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
gambasForms: Fix debug service image name

The docker-compose was not distinguishing the debug image from the
release image, so the debug image was being overwritten by the release
image. This commit fixes the image name for the debug service.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetAvalonia: Update to use dotnet-x-gtk base images

Does not make sense to use the horrible wayland base images since
Avalonia does not support wayland, so we are using the dotnet-x-gtk
base images that are more suitable for Avalonia applications.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetAvalonia: Fixes for the xfce healthcheck

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
scripts: Delete root package.json and package-lock.json

These does not make sense to be in the repository

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
scripts: connectDevice: Fix add a new device to the list of devices

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
scripts: cli: Add the device tree overlay commands

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
scripts: bash: Update the bash completion script

Now we have commands to list and apply Device Tree Overlays

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
scripts: Remove the core from the torizoncore-dev

Now it will be called torizon-dev, this is a breaking change, but it
was never released, so it is safe to do it now.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
scripts: zygote: add command not found message for target

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
script: zygote: make _target_apply_dto more CLI friendly

Let's accept the overlays as a parameter instead of using Read-Host

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
script: bash: Remove the core from the file name

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetAvaloniaFrameBuffer: Use commontorizon containers

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetAvaloniaFrameBuffer: fix framebuffer clear command

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetAvaloniaFrameBuffer: Use framebuffer backend as default for release

This change makes the framebuffer backend the default for the release
builds. This is because the DRM backend is not working properly on
Avalonia for all the hardware, and the framebuffer backend is more
stable and works on all the hardware.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetAvalonia: Dockerfile: Fix prod miss deps

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

Commit:
python3Console: Improve python related tasks

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
TASKS_CUSTOM_SETTINGS_JSON: Little fix

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
Dockerfile.sdk: Fix missing ARG IMAGE_ARCH on some templates

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
SSH_DEBUG_PORT change to DEBUG_SSH_PORT to be like the torizon_debug_ssh_port config

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
torizon_debug_port: Rename to torizon_debug_port1

torizon_debug_port: Rename to torizon_debug_port1, to better
standardize with torizon_debug_port2 and torizon_debug_port3.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
torizon_debug_ssh_port: Better organize position on settings.json

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

Commit:
tasks: Add DEBUG_SSH_PORT and DEBUG_PORTs where missing
Add DEBUG_SSH_PORT and DEBUG_PORTs where missing. Also, update
pull-container-... tasks to the same format as the others
(run-container-..., push-container-...).

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
docker-compose.yml: Update debug ports to be a settings.json value

docker-compose.yml: Update the debug and SSH debug ports to get them
from the corresponding settings.json value instead of having them
hardcoded on the docker-compose.yml file.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
cppQML and Slint: Organize a bit better debug ports on settings.json

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
dotnetAvalonia: Fix linux-amd64 build

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetAvaloniaFrameBuffer: Add ; true to fb0 clear command

The command to clear the framebuffer always will return an error,
so we need to add ; true to the end of the command to avoid
the task to fail.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
tasks: common: Add new task to open the project documentation

This task will open the project specific documentation in a new tab
without the need to open the project documentation file first.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetAvalonia: Automatically create preview assets on folder open

The Avalonia VS Code extension needs this to correctly show the preview

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
tasks: common: Change the pre-cleanup task behavior

The pre-cleanup task was changed to clean the service of the application
on the target (default device) and does not down the dependencies.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
tasks: common: Use the pre-cleanup-debug to stop only the app service

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
tasks: common: dotnetAvalonia: Experiment with rsync for deployment

The docker copy has two steps, compress and decompress, and it is
slow. This commit tries to use rsync to copy the files to the
Torizon device. The rsync command is added to the deploy-torizon-arm,
deploy-torizon-arm64, and deploy-torizon-amd64 tasks in the
dotnetAvalonia project.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetAvalonia: Fix the rsync from linux-x4 to linux-amd4

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: update packages versions

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

Commit:
Add script to check debug port and wait_sync conflics on multiContainer

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
dotnetUno5FrameBuffer: Add rsync deploy task and bump base image version

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: Add support for debugging on Torizon AMD64

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: add preferCSharpExtension to vscode settings

They continue to add configuration to use the dev kit instead of the
C# extension, but the C# extension is the recommended one for this one
at least. So, let's update it once more.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: Fix wrong base containers tag

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: add libfontconfig1 package as dependency

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: Update Uno packages to 5.3.114

This fixes the framebuffer rendering on Linux.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: add tasks to launch the app for Uno hot reload

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: add libinput10 package for fix input

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: Fix the launch task for hot reload

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: Automatically set the address for hot reload

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: Fix wrong task dependency

For the task `run-torizon-amd64` the dependency `deploy-torizon-amd64`
was removed because this task should be not architecture dependent.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: add console exit handler

When the user has a keyboard attached to the device, it is possible to
exit the application by pressing F12. But in a headless environment, it
is not possible to do that. This change adds a console exit handler to
the application that allows the user to exit the application by pressing
F12 in the debug terminal attached to the application stdout.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetAvalonia: add file nesting settings to vscode

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: prefer C# extension for dotnet projects instead dev kit

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: Fix the run for hot reload without the __change__

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: Add task to launch the application for hot reload at target

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

diff --git a/dotnetUno5/.vscode/tasks.json b/dotnetUno5/.vscode/tasks.json
index 41003bb..c0f3249 100644
--- a/dotnetUno5/.vscode/tasks.json
+++ b/dotnetUno5/.vscode/tasks.json
@@ -163,6 +163,74 @@
           "color": "terminal.ansiYellow"
       }
   },
+  {
+    "label": "run-app-for-hot-reload",
+    "hide": true,
+    "type": "shell",
+    "command": "ssh",
+    "presentation": {
+        "echo": true,
+        "reveal": "always",
+        "focus": false,
+        "panel": "dedicated",
+        "showReuseMessage": true,
+        "clear": true
+    },
+    "args": [
+        "-i",
+        "${workspaceFolder}/.conf/id_rsa",
+        "-p",
+        "${config:torizon_debug_ssh_port}",
+        "-o",
+        "UserKnownHostsFile=/dev/null",
+        "-o",
+        "StrictHostKeyChecking=no",
+        "-ttt",
+        "${config:torizon_run_as}@${config:torizon_ip}",
+        "cd ${config:torizon_app_root} && DOTNET_MODIFIABLE_ASSEMBLIES=debug UNO_FULLSCREEN=true ./__change__.Skia.Gtk",
+        "||",
+        "true"
+    ],
+    "dependsOrder": "sequence",
+    "problemMatcher": {
+        "pattern": {
+            "regexp": "^([^\\s].*)\\((\\d+|\\d+,\\d+|\\d+,\\d+,\\d+,\\d+)\\):\\s+(error|warning|info)\\s+(TS\\d+)\\s*:\\s*(.*)$",
+            "file": 1,
+            "location": 2,
+            "severity": 3,
+            "code": 4,
+            "message": 5
+        },
+        "background": {
+            "activeOnStart": true,
+            "beginsPattern": "^\\s*",
+            "endsPattern": "^\\s*"
+        }
+    },
+    "icon": {
+        "id": "flame",
+        "color": "terminal.ansiYellow"
+    }
+  },
+  {
+      "label": "launch-hot-reload-arm64",
+      "detail": "This task will launch the application on the board, without\nattach the debugger, this will allow the application to\nuse the Uno Platform hot reload feature.",
+      "command": "echo",
+      "type": "shell",
+      "args": [
+          "✅"
+      ],
+      "dependsOrder": "sequence",
+      "dependsOn": [
+          "config-hot-reload-address",
+          "deploy-torizon-arm64",
+          "run-app-for-hot-reload"
+      ],
+      "icon": {
+          "id": "vm-connect",
+          "color": "terminal.ansiYellow"
+      }
+  }
   ],
   "inputs": []
 }

microhobby committed:
dotnetUno5: tasks: Build the project on folder open

This will automatically generate the assets for xaml completion

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: Add build-and-reload-on-open task

This will aggregate the tasks that will first open the .csproj,
build the project and then reload the omnisharp server.
This way the use will have the completions for the xaml files
on the code behind files.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

diff --git a/dotnetUno5/.vscode/tasks.json b/dotnetUno5/.vscode/tasks.json
index badc339..2998f97 100644
--- a/dotnetUno5/.vscode/tasks.json
+++ b/dotnetUno5/.vscode/tasks.json
@@ -12,9 +12,29 @@
       "icon": {
           "id": "selection",
           "color": "terminal.ansiYellow"
+      }
+    },
+    {
+      "label": "build-and-reload-on-open",
+      "hide": true,
+      "command": "echo",
+      "type": "shell",
+      "args": [
+          "✅"
+      ],
+      "problemMatcher": "$msCompile",
+      "dependsOrder": "sequence",
+      "dependsOn": [
+        "open-skia-gtk-csproj",
+        "build-debug-local",
+        "reload-skia-gtk-csproj"
+      ],
+      "icon": {
+          "id": "loading",
+          "color": "terminal.ansiYellow"
       },
       "runOptions": {
-          "runOn": "folderOpen"
+        "runOn": "folderOpen"
       }
     },
     {
@@ -47,9 +67,6 @@
       "icon": {
         "id": "flame",
         "color": "terminal.ansiYellow"
-      },
-      "runOptions": {
-        "runOn": "folderOpen"
       }
     },
     {

microhobby committed:
dotnetUno5: Add task to run the app for hot reload locally

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

diff --git a/dotnetUno5/.vscode/tasks.json b/dotnetUno5/.vscode/tasks.json
index 2998f97..1bbebbe 100644
--- a/dotnetUno5/.vscode/tasks.json
+++ b/dotnetUno5/.vscode/tasks.json
@@ -183,6 +183,22 @@
           "color": "terminal.ansiYellow"
       }
   },
+  {
+    "label": "run-app-for-hot-reload-local",
+    "hide": true,
+    "type": "shell",
+    "command": "dotnet",
+    "options": {
+      "env": {
+        "DOTNET_MODIFIABLE_ASSEMBLIES": "debug"
+      }
+    },
+    "args": [
+        "run",
+        "--project",
+        "${workspaceFolder}/__change__.Skia.Gtk/__change__.Skia.Gtk.csproj",
+    ],
+  },
   {
     "label": "run-app-for-hot-reload",
     "hide": true,
@@ -232,6 +248,25 @@
         "color": "terminal.ansiYellow"
     }
   },
+  {
+    "label": "launch-hot-reload-local",
+    "detail": "This task will launch the application on the board, without\nattach the debugger, this will allow the application to\nuse the Uno Platform hot reload feature.",
+    "command": "echo",
+    "type": "shell",
+    "args": [
+        "✅"
+    ],
+    "dependsOrder": "sequence",
+    "dependsOn": [
+        "config-hot-reload-address",
+        "build-debug-local",
+        "run-app-for-hot-reload-local"
+    ],
+    "icon": {
+        "id": "vm-connect",
+        "color": "terminal.ansiYellow"
+    }
+  },
   {
       "label": "launch-hot-reload-arm64",
       "detail": "This task will launch the application on the board, without\nattach the debugger, this will allow the application to\nuse the Uno Platform hot reload feature.",

microhobby committed:
dotnetUno5: update nuget packages

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: tasks: Use x11 backend to load the hot reload app

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: Fix GDK_BACKEND value in launch.json and tasks.json

Let's use x11 until we have the https://github.com/unoplatform/uno/issues/17826

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: Update to use dotnet-gtk image

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: Fix Dockerfile to use GPU version of dotnet-gtk

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: downgrade Uno packages to 5.1.104

This fixes the GTK IME not being triggered

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: Use the latest version with Uno.UI.FeatureConfiguration

Setting the Uno.UI.FeatureConfiguration.TextBox.UseOverlayOnSkia to true
rollback the behavior of the GTK IME onboard keyboard to the previous
version, so we can use the latest version of Uno.WinUI.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: Rollback to the copy assets instead of docker COPY

The docker copy first compress the files and then decompress them
on the target, this makes the flow slower than the rsync command
that just copy the files directly to the target.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: tasks: Fix path to publish folder

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: Add Torizon AMD64 debug launch and deploy tasks

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

diff --git a/dotnetUno5/.vscode/launch.json b/dotnetUno5/.vscode/launch.json
index bd50b37..bae1a26 100644
--- a/dotnetUno5/.vscode/launch.json
+++ b/dotnetUno5/.vscode/launch.json
@@ -58,6 +58,41 @@
           "debuggerPath": "/vsdbg/vsdbg"
       },
       "preLaunchTask": "deploy-torizon-arm64"
+    },
+    {
+      "name": "Torizon AMD64",
+      "type": "coreclr",
+      "request": "launch",
+      "program": "${config:torizon_app_root}/__change__.Skia.Gtk",
+      "cwd": "${config:torizon_app_root}",
+      "stopAtEntry": false,
+      "console": "internalConsole",
+      "env": {
+        "DOTNET_MODIFIABLE_ASSEMBLIES": "debug",
+        //"GDK_BACKEND": "x11",
+        //"GDK_GL": "gles",
+        "UNO_FULLSCREEN": "true"
+      },
+      "args": [],
+      "pipeTransport": {
+          "pipeCwd": "${workspaceFolder}",
+          "pipeProgram": "ssh",
+          "pipeArgs": [
+              "-T",
+              "-q",
+              "-p",
+              "${config:torizon_debug_ssh_port}",
+              "-i",
+              "${workspaceFolder}/.conf/id_rsa", // ssh key path
+              "-o",
+              "StrictHostKeyChecking=no",
+              "-o",
+              "UserKnownHostsFile /dev/null",
+              "${config:torizon_run_as}@${config:torizon_ip}" // user@device
+          ],
+          "debuggerPath": "/vsdbg/vsdbg"
+      },
+      "preLaunchTask": "deploy-torizon-amd64"
     }
   ]
 }
diff --git a/dotnetUno5/.vscode/tasks.json b/dotnetUno5/.vscode/tasks.json
index 945ccf0..dd4b72a 100644
--- a/dotnetUno5/.vscode/tasks.json
+++ b/dotnetUno5/.vscode/tasks.json
@@ -97,6 +97,29 @@
           "color": "terminal.ansiCyan"
       }
     },
+    {
+      "label": "publish-debug-amd64",
+      "detail": "",
+      "command": "dotnet",
+      "type": "process",
+      "args": [
+          "publish",
+          "-c",
+          "Debug",
+          "${workspaceFolder}/__change__.Skia.Gtk/__change__.Skia.Gtk.csproj",
+          "/property:GenerateFullPaths=true",
+          "/consoleloggerparameters:NoSummary",
+          "-r",
+          "linux-amd64",
+          "--no-self-contained"
+      ],
+      "dependsOrder": "sequence",
+      "problemMatcher": "$msCompile",
+      "icon": {
+          "id": "flame",
+          "color": "terminal.ansiYellow"
+      }
+    },
     {
       "label": "publish-debug-arm64",
       "detail": "",
@@ -152,6 +175,38 @@
           "color": "terminal.ansiYellow"
       }
     },
+    {
+      "label": "deploy-torizon-amd64",
+      "detail": "",
+      "hide": true,
+      "command": "rsync",
+      "type": "process",
+      "args": [
+          "-av",
+          "-e",
+          "ssh -p ${config:torizon_debug_ssh_port} -i ${workspaceFolder}/.conf/id_rsa -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null",
+          "${workspaceFolder}/__change__.Skia.Gtk/bin/Debug/net8.0/linux-x64/publish/",
+          "${config:torizon_run_as}@${config:torizon_ip}:${config:torizon_app_root}"
+      ],
+      "dependsOn": [
+          "validate-settings",
+          "validate-arch-amd64",
+          "copy-docker-compose",
+          "pre-cleanup-debug",
+          "publish-debug-amd64",
+          "build-container-torizon-debug-amd64",
+          "push-container-torizon-debug-amd64",
+          "pull-container-torizon-debug-amd64",
+          "run-container-torizon-debug-amd64",
+          "pos-cleanup"
+      ],
+      "dependsOrder": "sequence",
+      "problemMatcher": "$msCompile",
+      "icon": {
+          "id": "flame",
+          "color": "terminal.ansiYellow"
+      }
+    },
     {
       "label": "clean-debug-local",
       "detail": "dotnet clean command for cleaning the local application debug build",

microhobby committed:
dotnetUno5: Add launch.json to the update.json file

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: Add task to launch app amd64 for hot reload

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cConsole: use commontorizon images

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cConsole: Use the rsync instead the docker COPY

This is more faster than the COPY command, docker first compress
the files and build the layers, that need to be uncompressed
when the container is pulled.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cConsole: fix rsync tasks to copy the content of the debug folder

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

Commit:
python3Pyside2QML: Add create-venv and improve python related tasks

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
rust and zig: Improve tasks presentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
monoCsharpForms: Improve tasks presentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
genericTemplate, java, gambas: Improve tasks presentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
node: Improve tasks presentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
common.json: Improve tasks presentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
python3: Improve tasks presentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
c/cpp: Improve tasks presentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
dotnet/aspnet: Improve tasks presentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
python3Console: Use commontorizon base and remove docker COPY

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
python3Console: Fix the requirements file copy and install

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
Revert "python3Console: Fix the requirements file copy and install"

This reverts commit f288f90934cb3ec8fec80fc33b1e2b33f6453f3a.

microhobby committed:
python3Console: Fix launch expected task name & remove pos-cleanup

For python sounds better to maintain the .venv folder in the container
and not remove it after the debug session, so the pos-cleanup task was
removed from the tasks.json file.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
python3Console: add stop-torizon-debug-generic task

The process need to be first stopped before try again to deploy the
application, so add a new task to stop the process on the target
device.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
python3Console: Make sure that the approot is owned by torizon

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
python3Console: Fix task to kill python3 process

The first time the task is executed, it will fail because the python3
process is not running.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
assets: tasks: common: Remove presentation settings from apply-packages

Using the same instance of the terminal where the pipeline is running
to execute the apply-packages task should speed up the process a bit.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

Commit:
tasks presentation: Group all build and execution tasks in a terminal

Group all build and execution tasks in a terminal, as opening a new
terminal decreases execution time.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
tasks: Set showReuseMessage: false for a cleaner output in terminal

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
tcb: tasks: Update duplicated tasks to the same as in common.json

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
cppConsole: Use the commontorizon and rsync instead COPY

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cppConsole: Fix the tag for the common sdk image

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
c: Always use the new GPU prefix for the Torizon containers

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

Commit:
projectUpdater:Add missing settings.json, launch.json and Dockerfile.sdk

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
cmakeConsole: Use commontorizon images

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cmakeConsole: Add rsync deploy tasks

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cmakeConsole: Fix Dockerfile.sdk to use commontorizon base image

The tag was updated but the base image was not updated to use the new
commontorizon organization.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cmakeConsole: fix rsync task to use correct build folder

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
monoCsharpForms: Update the commontorizon images & use rsync inst COPY

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
monoCsharpForms: Add xfce as default graphical compositor

Mono does not support wayland and never will, so we need to use X11
compositor to run graphical applications in a better way.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
javaForms: Use commontorizon base images

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
javaForms: User rsync instead of docker COPY

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
javaForms: Fix rsync-debug-dist task location

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

Commit:
update.json: Update the files of the templates

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
cppSlint: Fix the libslint_cpp.so path not found by ninja

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cppSlint: Use rsync instead docker COPY

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cppSlint: Dockerfile.debug: Fix base image tag

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cppSlint: Remove unused build stage for debug image

The libslint_cpp.so is really only needed for the build step

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
cppSlint: enable gpu prefix for get the new platform prefix

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetSlint: bump SlintDotnet to 1.7.1

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetSlint: Use commontorizon base images

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
templates: update dotnetSlint runtime to 1.7

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetSlint: Use rsync instead docker COPY on debug

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetSlint: Add a taks to refresh the projects

When using .NET Slint sometimes the code generator does not refresh the
projects, so it is necessary to restart the code generator to make it
work again.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
scripts: checkSingleProjectsConflicts.ps1: fix typo in message and args

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
cspell: Update word list

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
github: Run sanity checks on pull requests

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts/github: add script to check if all files have a new line at end

Now all the files should have a new line at the end, this script will
check if all the files have a new line at the end and if not it will
break the ci pipeline.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
all: Fix no new line at end of files

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
tcb: tcbuild: Updated it to use a quarterly

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
tcb: tasks: Add missing check-deps task

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
tasks: common: Drop the showReuseMessage and clear properties

This breaks the UX of the end of a task, so lets remove it.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
tcb: tasks: Drop showReuseMessage and clear from tasks.json

This breaks the UX for the tasks, as the output is not cleared and the
reuse message is not shown, this is not the expected behavior for the
tasks and appears that the tasks are still running when they already
finished.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

Commit:
checkSingleProjectsConflicts:Allow acceptAll arg, to bypass manual input

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
checkSingleProjectsConflicts: Fix working directory

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
all: tasks: Drop showReuseMessage and clear from tasks.json

This breaks the UX for the tasks, as the output is not cleared and the
reuse message is not shown, this is not the expected behavior for the
tasks and appears that the tasks are still running when they already
finished.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
all: tasks: Change all icon from red color to yellow

The red color brings an UX idea that something crashed or is wrong, so
change all icons from red to yellow to bring a more neutral color, or
like alert, to the tasks that are being executed.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
github: sanit-check: Fix powershell to pwsh

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
assets: common: tasks: Add debug ports to create prod image

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

Commit:
cppQML: Little fix on settings.json

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
common.json: Little fix on pre-cleanup(-debug) presentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

andreriesco commented on assets/tasks/common.json:
```diff
@@ -672,13 +731,157 @@
+            }
+        },
+        {
+            "label": "pre-cleanup-debug",
```
Tested here on imx6ull and it seems that it is not bringing down the other containers, which is leading to problems
![image](https://github.com/user-attachments/assets/4fad7b4e-7006-41b3-828b-873578380107)

microhobby replied:
This is by design, now it only down the service of the workspace application. This is yet another change to make the development flow faster between one deployment and another. What types of problems are you facing?

andreriesco replied:
It is not bringing down the other containers before running my container, slowing down the execution of the container

andreriesco replied:
It would be nice if we could figure out a way to run the previous command (which is called now `containers-cleanup` I think) at least the first time the person starts a debug session, so it starts with all other containers down and also removes the orphan containers

microhobby replied:
Please keep in mind that you are testing different stacks, this is not the field scenario where the user sticks in a workspace with a specific stack. 

Your user case is from a template developer/tester, so if run this makes sense to your scenario add it to the test pipeline.

tronical committed:
Update to Slint version 1.7.2

Signed-off-by: Simon Hausmann <simon.hausmann@slint.dev>

Commit:
python3Console: Improve deploy task to check if debug is ready

Improve deploy task to check if debug is ready, instead of waiting a
fixed time, improving the time to start debugging.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
python3Pyside2QML: Update tasks,launch matching the python3Console ones

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
projectUpdater:Don't allow proj update if there is no git repo initiated

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
Dockerfiles: Fix warning AS followed by a uppercase letter

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
Merge pull request #223 from andreriesco/release-2.6-prep

 checkSingleProjectsConflicts:Allow arg to bypass manual input, also allow execution from the IDE side. Python improve debug performance. Other fixes

microhobby committed:
all: Dockerfiles: Fix the missed stage names in lower case

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

Commit:
tasks: Update icon and presentation of tasks that were missing this

Also, remove duplicated pre-cleanup and pre-cleanup-debug.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

andreriesco committed:
Merge pull request #224 from slint-ui/dev

Update to Slint version 1.7.2

Commit:
tasks: Remove unused tasks

Criterial for removal: Tasks hidden (hide: true), which seem to not be
called anywhere in the extension's code or by any other task. Probably
old tasks that lost their usability with updates on other tasks or on
the extension's code.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
Merge pull request #225 from andreriesco/release-2.6-prep

tasks: Update icon and presentation of tasks that were missing this. Also remove unused tasks

Commit:
settings.json: Bump "tcb.version" setting to latest tcb version(3.10.0)

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
templates.json:Add deprecated:true to dotnetUno and dotnetUnoFrameBuffer

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
README: Add info about deprecated templates, on DEPRECATED.md

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
Merge pull request #226 from andreriesco/release-2.6-prep

Release 2.6 prep - minor updates

microhobby committed:
all: dockefiles: Use commontorizon base images

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
zigConsole: Dockerfile: Fix cross sdk base version

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
misc: Fix ci/cd pipeline Check EOF New Line

https://github.com/toradex/vscode-torizon-templates/actions/runs/10481960462/job/29032283020

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
assets: common: tasks: Add DOCKER_LOGIN variable to pull push release

Since we are using a local registry to build the release images to
test we need to also set the DOCKER_LOGIN variable to the local
registry IP address.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
scripts: projectUpdater: Remove repeated message input

Is not necessary to ask the user twice if he wants to proceed, it
makes it annoying.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
assets: common: tasks: The local registry should use the port 5002

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetConsole: Use rsync instead of Docker COPY

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

Commit:
templates.json: Remove deprecated dotnetUno and dotnetUnoFrameBuffer

Remove this deprecated templates from templates.json, so they don't
showup as options on project creation.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
createFromTemplate.ps1: Don't create torizonPackages.json for tcb

As tcb template doesn't have Dockerfiles, don't create
torizonPackages.json for tcb

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
Merge pull request #227 from andreriesco/release-2.6-prep

templates.json: Remove deprecated dotnetUno and dotnetUnoFrameBuffer.  Also, little fix on createFromTemplate.ps1

Commit:
tasks: Little fix on task icon color

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

andreriesco committed:
Merge pull request #228 from andreriesco/release-2.6-prep

tasks: Little fix on task icon color

Commit:
createFromTemplate: Fix comment, correct word

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

andreriesco committed:
Merge pull request #229 from andreriesco/release-2.6-prep

createFromTemplate: Fix comment, correct word

microhobby committed:
tcb: Fix platform push asking for docker credentials & add tcb.outputTEZIFolder

Related-to: TIE-922 TIE-906
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
gambasForms: remove vscode tasks.ps1

This file is copied from the scripts folder when the project is created

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: tasks: re-implement tcb.outputTEZIFolder

This is a re-implementation from the TCB extension, it reads the
tcbuild.yaml file and replaces the `${command:tcb.outputTEZIFolder}`
with the value from the yaml file.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
all: Dockerfiles: Bump all base images to 3.3.1

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: projectUpdater: fix extensions.json copy

Only try to copy the extensions.json file if it exists in the template
folder. This will prevent the script from failing when the file is not
present.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: projectUpdater: Fix script failing for templates without extensions.json

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: projectUpdater: Fix script asking duplicate confirmation

If the script update itself is called from the projectUpdater script, it
will ask for confirmation twice, this is because the script is called
twice, the first time to update the project and the second time to
update the projectUpdater script itself. This patch adds a new argument
to the script to avoid asking for confirmation on the second run.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

Commit:
dotnetAvaloniaFrameBuffer: Little update on one task presentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
common.json:Add debug ports env to build-container-torizon-release tasks

Even though this ports are not used on the release images, the build
fails with "no port specified: :<empty>" if those env vars are not there

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

andreriesco committed:
Merge pull request #230 from andreriesco/release-2.6-prep

Release 2.6 prep

Commit:
Debug ports args are also needed in the push, pull and run release tasks

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

andreriesco committed:
Merge pull request #231 from andreriesco/release-2.6-prep

Debug ports args are also needed in the push, pull and run release tasks

Commit:
pre-cleanup: Have just one pre-cleanup, that stops debug and release

Have just one pre-cleanup, that stops debug and release containers,
to avoid having the debug container and the release container running
at the same time.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
Merge pull request #232 from andreriesco/release-2.6-prep

pre-cleanup: Have just one pre-cleanup, that stops debug and release

Commit:
Fix "torizon.gpuPrefixRC": true for the remaining templates

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
Merge pull request #234 from andreriesco/release-2.6-prep

Fix "torizon.gpuPrefixRC": true for the remaining templates

microhobby committed:
all: Drop use of key pair for SSH debug

Related-to: #155
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
python: tasks: Use wait-a-bit task instead sleep

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
monoCsharpForms: add run-torizon-mono-debug-server-common task

This task will open a new terminal and run the mono debug server
showing the output in a dedicated panel.

Related-to: #176
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
ccpQML: Add exclusion for the build-* inside the github workflows

Related-to: #179
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
Dockerfiles: sdk: all: Always switch to torizon user

All the sdk Dockerfiles are now switching to the torizon user
before setting the working directory to the APP_ROOT.

Related-to: #186
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
assets: github/gitlab: Add call to docker-login task

This will fail fast in case of wrong credentials

Related-to: #188
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
rustSlint: update deploy tasks to use rsync instead Docker COPY

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
rustSlint: Rollback user change in Dockerfile.sdk

Seems that the rust does not liked the torizon user, so
I will rollback this change to avoid any issue.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
rustSlint: cargo: Bump slint to 1.7 and change backend to linuxkms

Related-to: #203
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
assets: gitlab: Fix vscode tasks path

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
assets: common: change icon for the task flame to account

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: tasks: Add edge case for config:docker_password

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
script: tasks: Fix special chars check to allow pipes

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
assets: common: Use command instead of input for docker login

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: projectUpdater: copy .dockerignore only if it exists

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: projectUpdater: fix trying to show merge diff for .dockerignore

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetUno5FrameBuffer: update to use dotnet-gtk base image

Event though the template uses framebuffer, it needs a lot of
dependencies to run that are already on the dotnet-gtk base image.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
Revert "dotnetUno5FrameBuffer: update to use dotnet-gtk base image"

This reverts commit df04e859d1bcd3cd9420f0df1e743022a42e75ef.

microhobby committed:
dotnetUno5FrameBuffer: add missing dependencies

Related-to: #236
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
monoCsharpForms: add missing dependencies

For debug we use the mono-complete, but for production no, so
we need to add the missing dependencies to the Dockerfile.

Related-to: #237
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

Commit:
templates.json: Missing "deprecated": false on zigConsole

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
zigConsole: Add C/C++ extension as recommended also

Add C/C++ extension as recommended also, as it uses GDB for debugging.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
nodeJSTypeScript: Fix tsc compilation error

Fixes #238

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
nodeJSTypeScript: Little fix on commit message

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
nodeJSTypeScript:Bump node version to 20, as it's the current active one

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
nodeElectron: Bump node version to 20, as it's the current active one

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
Merge pull request #239 from andreriesco/release-2.6-prep

node templates: Fixes and bump node version

microhobby committed:
javaForms: remove ARMv7 debug configuration

Since the Java appears to be abandoning 32bit architectures

https://openjdk.org/jeps/449

I was not able to find a JEP for the aarch32 deprecation, but
since 2021 there is not maintainance for the aarch32 port of
OpenJDK.

Related-to: #171
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetAvalonia: Dockerfile: Fix build for release

The solution file is needed to the Avalonia extension, so we need to
specify the project file to build the project.

Related-to: #240
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetAvaloniaFrameBuffer: fix build release

Related-to: #240
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
all: docker-compose: Allow access to tty devices

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
ccpQML: tasks: Use rsync to deploy the application instead Docker COPY

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
aspnetBlazor: update deploy tasks to use rsync

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
aspnetBlazor: tasks: Fix deploy tasks name convention

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
aspnetBlazor: Dockerfile: Fix release build

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
aspnetBlazor: tasks: Fix publish path for linux-amd64

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: node: Update the apollox lib

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: containers: Use commontorizon images

Repeat do not use torizonextras anymore!

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
all: binfmt: use commontorizon/binfmt image

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

Commit:
monoCsharpForms: Little fix on task

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
all: docker-compose: Bump weston image to 3.3.2

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
all: tasks: add --ansi never to docker compose commands

The ansi progress does not report the true progress of the docker
commands like pull push, so let's use the --ansi never to disable
the progress bar and show the real output of the commands.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
assets: common: tasks: Fix formatting issues

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

tronical committed:
Fix warning on startup about unrecognized renderer

> slint linuxkms backend: unrecognized renderer skia, falling back default

Enable the skia renderer to fix this.

Signed-off-by: Simon Hausmann <simon.hausmann@slint.dev>

andreriesco committed:
Merge pull request #246 from slint-ui/next

Fix warning on startup about unrecognized renderer

microhobby committed:
assets: tasks: common: Add --progress plain to docker build commands

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
assets: tasks: common: Fix progress plain to progress=plain

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
assets: tasks: common: Rollback to ansi to run-container-torizon-release

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
all: settings: Bump TCB version to v3.11.0

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
tasks: binfmt: Use commontorizon base image with the version 9.0.2

This make clear which version of qemu-user-static is being used

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
github-ci: use vscode tasks to run torizon binfmt

This make the consistency between the CI and the local development

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
all: tasks: Use -P option in rsync to show progress

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
all: tasks: Rollback --ansi never from docker compose up command

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

Commit:
python and node: Fix some tasks indentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
python: Add info about on .doc and unhide create-venv task

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
tcb: Add show-project-documentation task and initial tcb/.doc/README.md

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
rustSlint: Add build-debug-amd64 for remote target

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
rustSlint: fix build target

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
rustSlint: amd64: Fix target path to get the binary

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

Commit:
projectUpdater: Doc update for tcb template

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
tcb: tasks: Fix echo presentation in fill pipeline task

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: createDockerComposeProduction: Fixup for torizon_arch

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: createDockerComposeProduction: Fix error trhown for torizon_arch

This is not needed in the case of the createDockerComposeProduction
because there is cases where the right values will be set.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: tcb: Fix Github Actions dind workdir mount

When running the tcb in a docker-in-docker environment, the workdir
mount point is not found, because the github actions runner sandbox
the runner by re-writing the absolute path to a different location.
So, first we need to get the absolute path of the workspace and then
store it in a file called abs-path, so we can use get the path from
the file and use it as the workdir mount point.

Reported-by: @dk-coligo
Related-to: #202
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
scripts: projectUpdater: copy .doc folder instead of README.md

If the template had no README.md file in the .doc folder before
the projectUpdater script would fail to copy the README.md file
to the project folder. This commit changes the script to copy
the entire .doc folder instead of just the README.md file.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

Commit:
projectUpdater: Fix .doc copy

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

andreriesco committed:
Merge pull request #248 from andreriesco/release-2.6-prep

projectUpdater: Fix .doc copy

Commit:
projectUpdater: Better fix .doc copy

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
tcb: Add tcb-unpack as dependsOn of tcb-platform-push-ostree

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
tcb: Fix some hardcoded names, with __change__ and __container__

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
tcb: Avoid running tcb-unpack duplicated on the Github Action pipeline

As tcb-unpack is on the dependsOn of tcb-platform-push-ostree, removing
it from the Github Actions pipeline to avoid running it duplicated.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
tasks: Fix some tasks presentation

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby committed:
Merge pull request #249 from andreriesco/release-2.6-prep

projectUpdater: Better fix .doc copy and tcb template minor fixes
